source/beamline/SGM/monochromator/SGMGratingSupport.h

```diff
@@ -0,0 +1,140 @@
+#ifndef SGMGRATINGSUPPORT_H
+#define SGMGRATINGSUPPORT_H
+
+/*!
+  * Namespace containing enumerators and functions relating to the grating translation
+  * on the SGM Beamline
+  */
+namespace SGMGratingSupport {
+
+	/*!
+	  * Enumerates the different selections for the grating translation.
+	  */
+	enum GratingTranslation {
+		UnknownGrating = -1,
+		LowGrating = 0,
+		MediumGrating = 1,
+		HighGrating = 2
+	};
+
+	/*!
+	  * Spacing value for each of the grating translations used in the calculation
+	  * of the energy produced.
+	  * \param translationSelection ~ The grating whose spacing value is to be
+	  * returned.
+	  */
+	inline static double spacing(GratingTranslation translationSelection)
+	{
+		switch(translationSelection) {
+		case LowGrating:
+			return 1.689233e-6;
+		case MediumGrating:
+			return 9.163583e-7;
+		case HighGrating:
+			return 5.903703e-7;
+		default:
+			return 0;
+		}
+	}
+
+	/*!
+	  * Correction value used in the calculation of the energy produced by a given
+	  * grating translation.
+	  * \param translationSelection ~ The grating whose correction value is to be
+	  * returned.
+	  */
+	inline static double c1(GratingTranslation translationSelection)
```

Comment:
Yes! I have talked about this with Tom and he agrees. I need to get hold of
him for long enough for us to figure these out. In some cases, however,
these are just mathematical constructs (eg c1 and c2 are just correction
values).
On Sep 22, 2015 10:52 AM, "Darren Hunter" <notifications@github.com> wrote:

> In source/beamline/SGM/monochromator/SGMGratingSupport.h
> <https://github.com/acquaman/acquaman/pull/1620#discussion_r40111361>:
>
> > +		case MediumGrating:
> > +			return 9.163583e-7;
> > +		case HighGrating:
> > +			return 5.903703e-7;
> > +		default:
> > +			return 0;
> > +		}
> > +	}
> > +
> > +	/*!
> > +	  * Correction value used in the calculation of the energy produced by a given
> > +	  * grating translation.
> > +	  * \param translationSelection ~ The grating whose correction value is to be
> > +	  * returned.
> > +	  */
> > +	inline static double c1(GratingTranslation translationSelection)
>
> Are there more verbose names you could give these guys? I know they were
> probably called c1 and c2, but if they have a contextual name, I think that
> would benefit everyone (including our future selves when coming back to
> this code in the future).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/acquaman/acquaman/pull/1620/files#r40111361>.
>